src/ray/core_worker/experimental_mutable_object_manager.cc

```diff
@@ -382,6 +386,84 @@ Status MutableObjectManager::ReadAcquire(const ObjectID &object_id,
   return Status::OK();
 }
 
+Status MutableObjectManager::Wait(const std::vector<ObjectID> &object_ids,
+                                  int num_objects,
+                                  int64_t timeout_ms,
+                                  std::vector<bool> *results) {
+  RAY_LOG(DEBUG) << "MutableObjectManager::Wait " << object_ids.size()
+                 << ", num_objects: " << num_objects << ", timeout_ms: " << timeout_ms;
+  absl::ReaderMutexLock guard(&destructor_lock_);
+  RAY_LOG(DEBUG) << "MutableObjectManager::Wait get destructor lock";
+  std::vector<Channel *> channels;
+  for (const auto &object_id : object_ids) {
+    Channel *channel = GetChannel(object_id);
+    if (!channel) {
+      return Status::ChannelError("Channel has not been registered");
+    }
+    channels.push_back(channel);
+  }
+
+  bool done = false;
+  bool timed_out = false;
+  Status signal_status = Status::OK();
+  int64_t remaining_timeout = timeout_ms;
```

Comment:
figure out timeout

Reply:
only driver will set timeout. if it is running on the actors, it is infinite (10^6 sec).